modules/photons_control/clean.py

```diff
@@ -0,0 +1,188 @@
+from photons_app.errors import PhotonsAppError
+
+from photons_messages import LightMessages, DeviceMessages
+
+
+def SetCleanConfig(indication=None, duration=None, **kwargs):
+    """
+    Returns a valid message that will either get or set the default clean cycle
+    configuration for the device.
+
+    For example:
+
+    .. code-block:: python
+
+        await target.send(CleanConfig(indication=True), ["d073d5000001", "d073d5000001"])
+    """
+
+    async def gen(reference, sender, **kwargs):
+        has_hev = DeviceMessages.GetVersion()
+        async for pkt in sender(has_hev, reference, **kwargs):
+            if pkt | DeviceMessages.StateVersion:
+                if pkt.product == 90 or pkt.product == 99:
+                    yield LightMessages.SetHevCycleConfiguration(
+                        indication=True if indication else False,
+                        duration_s=duration if duration else 7200,
+                        target=pkt.serial,
+                        res_required=False,
+                    )
+
+    return FromGenerator(gen)
+
+
+@task
+class set_clean_default_config(task.Task):
+    """
+    Set the default clean cycle configuration for a device
+
+    ``target:clean_config match:cap=hev -- '{"indication": true, "duration": 7200}'``
+
+    It takes an "indication" field that is a boolen which indicates whether to
+    run a short flashing indication at the end of the HEV cycle. It also takes
+    a ``duration`` field that is the seconds of the duration.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = self.photons_app.extra_as_json
+
+        if "indication" not in options and "duration" not in options:
+            raise PhotonsAppError("You must specify a new indication or duration value.")
+
+        if "indication" not in options:
+            options["indication"] = False
+
+        if "duration" not in options:
+            options["duration"] = 7200
+
+        await self.target.send(CleanConfig(**options), self.reference, **kwargs)
+
+
+@task
+class get_clean_default_config(task.Task):
+    """
+    Get the default HEV cleaning cycle configuration for the target bulb
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        async def gen(reference, sender, **kwargs):
+            ps = sender.make_plans("capability")
+            async for serial, _, info in sender.gatherer.gather(ps, reference, **kwargs):
+                if info["cap"].has_hev:
+                    yield LightMessages.GetHevCycleConfiguration(target=serial)
+
+        async for pkt in self.target.send(FromGenerator(gen), self.reference):
+            print(pkt.serial)
+            indication = "enabled" if pkt.indication else "disabled"
+            print(f"\tCycle time: {pkt.duration_s}")
```

Comment:
```suggestion
            print(f"   Cycle time: {pkt.duration_s}")
```

I used to use tabs everywhere and then there was a point I got annoyed by them and started trying to make them explicit spaces.

Reply:
I was actually surprised that responses/output are not structured by default. Shows you how often I use `lifx` the CLI.